Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -1,37 +1,329 @@
+                }
+            }
+            return highestBalanceEver;
+        }
+
+        public static float GetHighestCurrentBalance(string[] peopleAndBalances)
+        {
+            string[][] workingPeopleAndBalances = ArrayifyBalances(peopleAndBalances);
+            float highestCurrentBalance = float.Parse(workingPeopleAndBalances[0][^1]);
+            if (workingPeopleAndBalances.Length == 1)
+            {
+                return highestCurrentBalance;
+            }
+            for (int i = 1; i < workingPeopleAndBalances.Length; i++)
+            {
+                float balanceToTest = float.Parse(workingPeopleAndBalances[i][^1]);
+                if (balanceToTest > highestCurrentBalance)
+                {
+                    highestCurrentBalance = balanceToTest;
+                }
+            }
+            return highestCurrentBalance;
+        }
+
+        public static float GetLowestCurrentBalance(string[] peopleAndBalances)
+        {
+            string[][] workingPeopleAndBalances = ArrayifyBalances(peopleAndBalances);
+            float lowestCurrentBalance = float.Parse(workingPeopleAndBalances[0][^1]);
+            if (workingPeopleAndBalances.Length == 1)
+            {
+                return lowestCurrentBalance;
+            }
+            for (int i = 1; i < workingPeopleAndBalances.Length; i++)
+            {
+                float balanceToTest = float.Parse(workingPeopleAndBalances[i][^1]);
+                if (balanceToTest < lowestCurrentBalance)
+                {
+                    lowestCurrentBalance = balanceToTest;
+                }
+            }
+            return lowestCurrentBalance;
+        }
+
+        /// <summary>
+        /// Takes a balance then finds and returns all the names that share that balance
+        /// </summary>
+        /// <param name="balance"></param>
+        /// <param name="peopleAndBalances"></param>
+        /// <returns></returns>
+        public static string[] FindBalanceNames(float balance, string[]peopleAndBalances, bool isCurrent)
+        {
+            string[][] workingPeopleAndBalances = ArrayifyBalances(peopleAndBalances);
+            string totalNames = "";
+            for (int i = 0; i < workingPeopleAndBalances.Length; i++)
+            {
+                float[] balances = ConvertStringArrToFloatArr(workingPeopleAndBalances[i][1..]);
+                bool isBalance = false;
+                int isCurrentIndexShift = 0;
+                if (isCurrent)
+                {
+                    isCurrentIndexShift = balances.Length - 1;
+                }
+                for (int j = 0 + isCurrentIndexShift; j < balances.Length; j++)
+                {
+                    if(balances[j] == balance)
+                    {
+                        isBalance = true;
+                    }
+                }
+                if (isBalance)
+                {
+                    totalNames += $"{workingPeopleAndBalances[i][0]}, ";
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use a StringBuilder instead.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5156603)